backend/utils/deployShop.js

```diff
@@ -196,13 +196,19 @@ async function _deployShop({
       : 'localhost'
 
   function replaceVars(html) {
-    return html
-      .replace('TITLE', publicShopConfig.fullTitle)
-      .replace('META_DESC', publicShopConfig.metaDescription || '')
-      .replace('DATA_DIR', dataDir)
-      .replace(/NETWORK/g, networkName)
-      .replace('FAVICON', publicShopConfig.favicon || 'favicon.ico')
-      .replace('UI_SRC', networkConfig.uiCdn || '')
+    return (
+      html
+        .replace('TITLE', publicShopConfig.fullTitle)
+        .replace('META_DESC', publicShopConfig.metaDescription || '')
+        .replace('DATA_DIR', dataDir)
+        .replace(/NETWORK/g, networkName)
+        .replace('FAVICON', publicShopConfig.favicon || 'favicon.ico')
+        .replace('UI_SRC', networkConfig.uiCdn || '')
+        // Add GA code just before </head> tag
+        // TODO: Technically any script can be injected,
+        // make sure to add a warning in frontend
+        .replace('</head>', `${publicShopConfig.gaCode || ''}</head>`)
```

Comment:
@nick @gullible1 Do we want to do add a warning to the frontend? Could open up XSS vulnerabilities if user pastes scripts from suspicious sites

Reply:
I'm thinking we use the same mechanism as the Admin for GA. `See shop/src/pages/App.js` line 92.

Reply:
Have made that change, thanks for pointing it out